### ISSUE TITLE
[ENHANCEMENT] - Branch highlight focus in history graph

### ISSUE THREAD

**Issue by Shimoon** [sourcegit-scm/sourcegit#253]
### Problem
When a repository becomes bigger, it gets hard to follow which commits belong to which branch, which branch gets merged into which one or the impact of one branch over the global scope. Right now, the only solution to reduce the visual clutter is to filter the wanted branch, but in this case, we loose the global repository overview and the interactions of that particular branch over the other ones.

### Proposal
I suggest a toggle option to only color the current branch from HEAD or visual selection and all the ancestors while the other branches becomes dimmed.
Like this, we could clearly track which commits are involved into the current branch state while keeping a good overview of the whole repository.

**Comment by love-linger** [comment]
![image](https://github.com/sourcegit-scm/sourcegit/assets/4075673/ed8b8703-a331-4874-ab7f-035bd2feed6e)

**Comment by gadfly3173** [comment]
@love-linger In some cases, the current commit highlighting may not highlight all parent commits.

![image](https://github.com/sourcegit-scm/sourcegit/assets/28685179/d2ef4b92-983f-41a7-9ded-2a0c1e210320)
![image](https://github.com/sourcegit-scm/sourcegit/assets/28685179/65d46355-5f47-4f22-8b11-5952ba14d5d1)

If the current HEAD has multiple parent commits, it may happen that only one parent commit and its ancestor nodes will be colored.

**Comment by love-linger** [comment]
I've push a commit that try to fix this issue. @gadfly3173 could you test it for me?

**Comment by gadfly3173** [comment]
> I've push a commit that try to fix this issue. @gadfly3173 could you test it for me?

LGTM
